WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt

```diff
@@ -341,10 +342,28 @@ class WPMainActivityViewModelTest : BaseUnitTest() {
 
     @Test
     fun `when user taps to open the login page from the bottom sheet empty view cta the correct action is triggered`() {
+        var loginFlowTriggered = false
+        var switchTabTriggered = false
+
+        viewModel.startLoginFlow.observeForever { event ->
+            event.applyIfNotHandled {
+                loginFlowTriggered = true
+            }
+        }
+
+        viewModel.switchToMySite.observeForever { event ->
+            event.applyIfNotHandled {
+                switchTabTriggered = true
+            }
+        }
+
         startViewModelWithDefaultParameters()
-        viewModel.onOpenLoginPage()
 
-        assertThat(viewModel.startLoginFlow.value!!.peekContent()).isTrue()
+        viewModel.onOpenLoginPage(0)
+
+        assertThat(loginFlowTriggered).isTrue
+        verify(appPrefsWrapper, times(1)).setMainPageIndex(eq(0))
```

Comment:
🔍 Nitpick: I realize this test was not introduced in this PR, but since we are adding more assertions I think the comment is still relevant. Having said that feel free to ignore it.

Usually it's better to have a single focused tests with a clear name explaining what they are testing. There are two good reasons why this practice can be useful.
1. Unit tests serve as a documentation for other devs (future-self) -> "the correct action is triggered" doesn't say much -> the documentation should say what the "correct actions" are.
2. Unit tests would ideally have a single reason to fail -> when it fails anyone should be able to quickly find the issue and fix it without (almost any) domain knowledge.
3. 
We might want to consider splitting this into 3 test cases
1. "...., the login page flow is started"
2. "..., the default bottom sheet page is set to MySite"
3. "..., the bottom sheet tab is switched to MySite"

Let me know what you think ;)

Reply:
makes sense 👍 ! Changed to follow your advice. Thanks for all the feedbacks @malinajirka 🙇 !